### PR TITLE
Add a .node-version file to document supported node version

### PR DESCRIPTION
**Pull request by mcintyre94**
Since there's some issue with a dependency on v18, hopefully this will save a bit of time next time we need to build it! `.node-version` seems to be the closest to a standard supported by a bunch of tools, but is probably mostly just useful in our case for documentation.

### CODE REVIEW

File: .node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+16.19.0
```
Comment (jordaaash) [inline .node-version]: Does this need to be a specific pinned version? Could it be like `16.x`?

File: README.md [hunk 2]
```diff
@@ -247,6 +247,8 @@ extension|string (`'png' 'jpeg' 'webp' 'svg'`)|`'png'`      |File extension
 
 If you get an error running `npm install` referring to `node-pre-gyp`, this is caused by an attempt to compile the [`canvas` dependency](https://github.com/Automattic/node-canvas#compiling). See Compiling instructions in the README. For example on MacOS you need to install dependencies: `brew install pkg-config cairo pango libpng jpeg giflib librsvg pixman`.
 
+Currently this repo will not build (`npm run build`) on Node v18, recommended version is v16.
```
Comment (jordaaash) [inline README.md]: Can we add a link or something (either here or in this issue) related to why it won't work on Node 18?

**Review by jordaaash — COMMENTED** [review]
Makes sense!

**Comment by jordaaash** [comment]
Actually, can we (maybe also) enforce this using the `engines.node` field in the package.json?

https://render.com/docs/node-version

File: .node-version (hunk 1 above)
Comment (mcintyre94) [reply .node-version]: Annoyingly this isn't really standardised, but it seems [most tools support partial versions](https://github.com/shadowspawn/node-version-usage#compatibility-testing). I've just changed this to `16`, which I think is more useful/less misleading from a documentation standpoint

File: README.md (hunk 2 above)
Comment (mcintyre94) [reply README.md]: Good shout, added a link to https://stackoverflow.com/q/69692842/1375972 

There are some other answers there pointing to solutions to fix it on v18. I was going to open an issue to come back to that at some point but don't seem to be able to open issues in this repo. But we've got the link here + in the readme for now

**Comment by mcintyre94** [comment]
> Actually, can we (maybe also) enforce this using the `engines.node` field in the package.json?
> 
> https://render.com/docs/node-version

Neat, TIL! Added this

Seems to only be checked on `npm install`, but you get a pretty useful message:

```bash
$ npm install
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@solana/qr-code-styling@1.6.0',
npm WARN EBADENGINE   required: { node: '>=16, <17' },
npm WARN EBADENGINE   current: { node: 'v18.16.0', npm: '9.5.1' }
npm WARN EBADENGINE }
```

Seems to [be advisory only by default](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#engines) for now, but should be the best place to enforce it for tools

File: package.json [hunk 3]
```diff
@@ -10,6 +10,9 @@
   "files": [
     "lib"
   ],
+  "engines": {
+    "node": ">=16, <17"
```
Comment (jordaaash) [inline package.json]: nit:

```suggestion
    "node": "16.x"
```

**Comment by jordaaash** [comment]
Thanks!

**Pull request merged**